src/ShortcutConfigurator.cc

```diff
@@ -0,0 +1,454 @@
+                        if (!QString::compare(shortcut, "DEFAULT", Qt::CaseInsensitive))
+                        {
+                            if(shortCut==QString::fromUtf8("")) continue;
+                            shortcutOccupied.insert(shortCut,action);
+                            shortcutsListFinal.append(shortCut);
+                        }
+                        else 
+                        {
+                            if(shortcutOccupied[shortcut])
+                            {
+                                raiseError(QString::fromLocal8Bit(configFileLoc.c_str())+QString(":\n"+actionName+" shortcut \""+shortcut+"\" conflicts with "+shortcutOccupied[shortcut]->objectName()));
+                                return;
+                            }
+                            else if(shortcut!=QString::fromUtf8("")) 
+                            {
+                                shortcutOccupied.insert(shortcut,action);
+                            }
+                            shortcutsListFinal.append(shortcut);
+                        }
+                    }
+
+                    action->setShortcuts(shortcutsListFinal);
+                }   
+            }
+        }
+    }
+
+}
+
+QString ShortcutConfigurator::getData(int row,int col)
+{
+    return shortcutsTable->model()->index(row,col).data().toString();
+}
+
+void ShortcutConfigurator::putData(QModelIndex indexA,QString data)
+{
+    shortcutsTable->model()->setData(indexA, data);
+}
+
+void ShortcutConfigurator::raiseError(const QString errorMsg)
+{
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error",errorMsg);
+        messageBox.setFixedSize(500,200);
+}
+
+void ShortcutConfigurator::onTableCellClicked(const QModelIndex & index)
+{
+    if (index.isValid() && index.column()!=0) 
+    {
+        //just to avoid opening of multiple dialogs on multiple clicks on cell
+        if(shortcutCatcher) delete shortcutCatcher;
+
+        shortcutCatcher = new QMessageBox;
+        QString updatedAction = getData(index.row(),0);
+        auto itr = shortcutsMap.find(updatedAction);
+        QAction* changedAction = itr.value();
+
+        shortcutCatcher->raise();
+        shortcutCatcher->setStyleSheet("QLabel{min-width: 400px;}");
+        shortcutCatcher->setInformativeText(QString(""));
+        shortcutCatcher->setWindowTitle(QString("Set Shortcut for: ")+updatedAction);
+        shortcutCatcher->installEventFilter(this);
+        shortcutCatcher->setText("Press the Key Sequence");
+        shortcutCatcher->setStandardButtons(QMessageBox::Apply | QMessageBox::Reset | QMessageBox::Cancel);
+        shortcutCatcher->setDefaultButton(QMessageBox::Apply);
+        shortcutCatcher->setWindowModality(Qt::WindowModal);
+        int ret = shortcutCatcher->exec();
+        QString updatedShortcut;
+
+        switch (ret)
+        {
+            case QMessageBox::Apply:
```

Comment:
What should we expect to happen if a user clicks Apply without entering a keypress? Can we disable the button until a keypress is given?

I found that since `pressedKeySequence` is kept alive as a member of `ShortcutConfigurator`, it currently will try to apply the last used keypress again (assuming a keypress was already given for any other action).  At the very least I would suggest clearing to an empty keypress before returning from this function, by using the default constructor: `pressedKeySequence = QKeySequence();`